webrecorder/browsertrix
# fix: Fix collection description

Fixes https://github.com/webrecorder/browsertrix/issues/2064

### Changes

- Switches MDE library to one that supports shadow DOM
- Refactors collection components to btrix components
- Fixes collection detail not expanding and contracting correctly

### Manual testing

1. Log in as crawler
2. Go to Collections
3. Click "New Collection"
4. Enter in name and description. Verify description can be edited as markdown as expected
5. Save
6. Scroll down to description. Verify description renders as expected
7. Click edit pencil button
8. Update description. Verify it saves as expected

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Collection Metadata Dialog | <img width="757" alt="Screenshot 2024-09-04 at 11 51 19 AM" src="https://github.com/user-attachments/assets/51c9fc5e-10b5-4e8c-85dd-1ce17fc3e4a8"> |


### Context

The initial issue with wysimark was the styles were no longer applied, even though the btrix markdown editor renders in light dom. I fixed the issue but found additional issues with cursor location and text selection.

It doesn't look like wysimark is maintained or fixing shadow dom issues anytime soon so I tried out https://github.com/davidmyersdev/ink-mde. It works well with shadow dom and has recent releases.
